src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java

```diff
@@ -0,0 +1,145 @@
+    fedoraUser = username;
+    fedoraPass = password;
+  }
+
+  /**
+   * Overloaded password authenticator.
+   *
+   * @return PasswordAuthentication
+   *   Authentication for this authenticator.
+   */
+  protected PasswordAuthentication getPasswordAuthentication() {
+    return new PasswordAuthentication(fedoraUser, fedoraPass.toCharArray());
+  }
+
+}
+
+/**
+ * Utilities for interfacing with Fedora in ways that GSearch can't.
+ */
+public class FedoraUtils {
+
+  protected static final Logger logger = Logger.getLogger(FedoraUtils.class);
+
+  /**
+   * Gets an InputStream for a datastream.
+   *
+   * @param pid
+   *   The PID of the object that has the datastream.
+   * @param dsId
+   *   The ID of the datastream to get.
+   * @param fedoraBase
+   *   The base URL of fedora, including protocol; e.g., 
+   *   http://localhost:8080/fedora.
+   * @param fedoraUser
+   *   The Fedora username to connect with.
+   * @param fedoraPass
+   *   The Fedora password to connect with.
+   *
+   * @return InputStream
+   *   An InputStream at the constructed dissemination point.
+   */
+  public static InputStream getDatastreamDisseminationInputStream(String pid, String dsId, String fedoraBase, String fedoraUser, String fedoraPass) {
+    URL url;
+    // Attempt to generate the URL from input.
+    try {
+      url = getDatastreamDisseminationURL(pid, dsId, fedoraBase, fedoraUser, fedoraPass);
+      return url.openStream();
+    }
+    // On exception, log and return a stream with no content so the caller
+    // doesn't get messed up.
+    catch (MalformedURLException e) {
+      logger.warn(String.format("Attempt to generate URL for datastream dissemination failed: %s", e.getMessage()));
+    }
+    catch (IOException e) {
+      logger.warn(String.format("Failed to open stream: %s", e.getMessage()));
+    }
+    return new ByteArrayInputStream("".getBytes());
+  }
+
+  /**
+   * Gets the raw text of a datastream.
+   *
+   * @param pid
+   *   The PID of the object that has the datastream.
+   * @param dsId
+   *   The ID of the datastream to get.
+   * @param fedoraBase
+   *   The base URL of fedora, including protocol; e.g.,
+   *   http://localhost:8080/fedora.
+   * @param fedoraUser
+   *   The Fedora username to connect with.
+   * @param fedoraPass
+   *   The Fedora password to connect with.
+   *
+   * @return String
+   *   The text of the given datastream.
+   */
+  public static String getRawDatastreamDissemination(String pid, String dsId, String fedoraBase, String fedoraUser, String fedoraPass) {
+    try {
+      URL url = getDatastreamDisseminationURL(pid, dsId, fedoraBase, fedoraUser, fedoraPass);
+      InputStream dsStream = url.openStream();
+      Scanner scanner = new Scanner(dsStream);
+      // If no content, return an empty string.
+      String dsString = scanner.hasNext() ? scanner.next() : "";
+      logger.debug(String.format("getRawDatastreamDissemination (pid: %s, DSID: %s): %s", pid, dsId, dsString));
```

Comment:
Should the `logger.warn()` calls that use `String.format()` similarly incorporate `logger.isWarnEnabled()`?